index.rst

```diff
@@ -1,61 +1,338 @@
+    By default, user-generated access tokens do not expire, although their permisisons are tied to the user's own permissions and thus they may become unusuable if the account is frozen or deleted or its access permissions change.
+
+IDM-0103
+    Internal Science Platform components may also generate temporary access tokens to facilitate multi-layer services.
+    These tokens must also be shown on the user's token list and the user must be able to revoke them, but they should be visually separated from user-created access tokens.
+    Internal temporary access tokens must expire in a reasonable length of time, such as shortly after the expected maximum duration of the operation for which they were intended.
+
+IDM-0104
+    Tokens should be scoped to restrict their power.
+    However, the number of scopes should not be so large as to be overwhelming.
+    A user should be able to easily choose the necessary scope of a token for common token-based workflows.
+    User-visible scopes should be limited to at most a few dozen, preferrably fewer.
+
+IDM-0105
+    Tokens must not contain a frozen representation of group membership or permissions.
+    Updates to the group membership of a user's account should also apply to all tokens issued for that user, provided that the scope of the token allows access.
+    See `SQR-039 <https://sqr-039.lsst.io/>`__ for more discussion.
+
+Logging
+-------
+
+IDM-0200
+    All initial authentications must be logged.
+    The log must include the external IP address of the authenticating client, information about the identity provided by the identity provider, and the mapped Science Platform account (if any).
+
+IDM-0201
+    All changes to the authentication metadata, such as changes to federated identity bindings, must be logged.
+
+IDM-0202
+    All token authentications from outside the Science Platform must be logged.
+
+IDM-0203
+    Users must be able to see their recent web authentications, at least including timestamp and external authentication provider.
+    Ideally this should include GeoIP information for the IP address, although getting accurate data inexpensively can be challenging so this isn't a firm requirement.
+
+IDM-0204
+    When displaying the list of federated identities associated with the account, the date and time at which that identity was last used to authenticate must be shown alongside.
+
+IDM-0205
+    When displaying the list of user-generated tokens, the date and time at which a user-generated token was last used must be shown alongside the token name.
+
+IDM-0206
+    Users must be notified via email of any change to their linked federated identities or any creation or deletion of a new user-generated token.
+
+Account management
+==================
+
+Status
+------
+
+IDM-1000
+    Accounts that are pending approval (under IDM-0003) can authenticate and see their account status and metadata page, but not access any other part of the Science Platform.
```

Comment:
so does this mean they can get tokens?  It seems like we wouldn't want to issue tokens until they have an account?  Or are you thinking there would be a capability like preapproved or just a token with no claims?

Reply:
Good catch. I've explicitly added that pending and frozen accounts may not create tokens and existing tokens for those accounts must not be accepted for authentication.